### ISSUE TITLE
"TypeError: e.getMetrics is not a function" when trying to update HPA to v2beta2

### ISSUE THREAD

**Issue by RotemBirman** [kontena/lens#3797]
**Describe the bug**
When trying to update HPA from lens to v2beta2 we get "TypeError: e.getMetrics is not a function" 

**To Reproduce**
Scenario 1 - 
1. Open Lens and got to HPA
2. Try to change HPA to v2beta2 with correct schema of metrics
3. See HPA page crash + "TypeError" 
==================
Scenario 2 - 
1. Update HPA to v2beta2 using kubectl or rancher
2. Go to lens -> HPA 
3. See that Lens is showing version v2beta1 and old schema.

**Expected behavior**
Scenario 1 - HPA will be updated with no  "TypeError: "
Scenario 2 - Lens will disaply the correct details of the HPA

**Screenshots**
![rancher](https://user-images.githubusercontent.com/38754536/133309251-e9e9b143-b26d-441c-93d6-6df155146211.png)
![Lens](https://user-images.githubusercontent.com/38754536/133309255-2486a55b-02fe-45c7-b370-bb185e502915.png)

**Environment (please complete the following information):**
- Lens Version: 4.2.5
- OS: Ubuntu 20.04.3 LTS
- Installation method (e.g. snap or AppImage in Linux): Snap

**Logs:**
When you run the application executable from command line you will see some logging output. Please paste them here:
```
Error stack:

 
TypeError: e.getMetrics is not a function
    at lj.getTargets (http://477ce64e-8ed9-43a0-bbc7-2b8b91697515.localhost:36695/build/Lens.js:2:8351649)
    at renderTableContents (http://477ce64e-8ed9-43a0-bbc7-2b8b91697515.localhost:36695/build/Lens.js:2:8352895)
    at ny.getRow (http://477ce64e-8ed9-43a0-bbc7-2b8b91697515.localhost:36695/build/Lens.js:2:8117133)
    at http://477ce64e-8ed9-43a0-bbc7-2b8b91697515.localhost:36695/build/Lens.js:2:7963886
    at http://477ce64e-8ed9-43a0-bbc7-2b8b91697515.localhost:36695/build/Lens.js:2:7702479
    at http://477ce64e-8ed9-43a0-bbc7-2b8b91697515.localhost:36695/build/Lens.js:2:7702092
    at ce (http://477ce64e-8ed9-43a0-bbc7-2b8b91697515.localhost:36695/build/Lens.js:2:12079)
    at e.track (http://477ce64e-8ed9-43a0-bbc7-2b8b91697515.localhost:36695/build/Lens.js:2:23443)
    at http://477ce64e-8ed9-43a0-bbc7-2b8b91697515.localhost:36695/build/Lens.js:2:7702068
    at http://477ce64e-8ed9-43a0-bbc7-2b8b91697515.localhost:36695/build/Lens.js:2:7701905
```
====
```
Component stack:


    at http://477ce64e-8ed9-43a0-bbc7-2b8b91697515.localhost:36695/build/Lens.js:2:7702458
    at div
    at div
    at t (http://477ce64e-8ed9-43a0-bbc7-2b8b91697515.localhost:36695/build/Lens.js:2:7954473)
    at div
    at qh (http://477ce64e-8ed9-43a0-bbc7-2b8b91697515.localhost:36695/build/Lens.js:2:7962442)
    at div
    at ip (http://477ce64e-8ed9-43a0-bbc7-2b8b91697515.localhost:36695/build/Lens.js:2:7964625)
    at div
    at div
    at ny (http://477ce64e-8ed9-43a0-bbc7-2b8b91697515.localhost:36695/build/Lens.js:2:8114932)
    at Ay (http://477ce64e-8ed9-43a0-bbc7-2b8b91697515.localhost:36695/build/Lens.js:2:8124250)
    at lj (http://477ce64e-8ed9-43a0-bbc7-2b8b91697515.localhost:36695/build/Lens.js:2:8351597)
    at t (http://477ce64e-8ed9-43a0-bbc7-2b8b91697515.localhost:36695/build/Lens.js:2:227828)
    at t (http://477ce64e-8ed9-43a0-bbc7-2b8b91697515.localhost:36695/build/Lens.js:2:229908)
    at cc (http://477ce64e-8ed9-43a0-bbc7-2b8b91697515.localhost:36695/build/Lens.js:2:7832442)
    at main
    at div
    at http://477ce64e-8ed9-43a0-bbc7-2b8b91697515.localhost:36695/build/Lens.js:2:7702458
    at gR (http://477ce64e-8ed9-43a0-bbc7-2b8b91697515.localhost:36695/build/Lens.js:2:8382221)
    at t (http://477ce64e-8ed9-43a0-bbc7-2b8b91697515.localhost:36695/build/Lens.js:2:227828)
    at t (http://477ce64e-8ed9-43a0-bbc7-2b8b91697515.localhost:36695/build/Lens.js:2:229908)
    at cc (http://477ce64e-8ed9-43a0-bbc7-2b8b91697515.localhost:36695/build/Lens.js:2:7832442)
    at main
    at div
    at pU (http://477ce64e-8ed9-43a0-bbc7-2b8b91697515.localhost:36695/build/Lens.js:2:8663604)
    at cc (http://477ce64e-8ed9-43a0-bbc7-2b8b91697515.localhost:36695/build/Lens.js:2:7832442)
    at t (http://477ce64e-8ed9-43a0-bbc7-2b8b91697515.localhost:36695/build/Lens.js:2:224373)
    at BU (http://477ce64e-8ed9-43a0-bbc7-2b8b91697515.localhost:36695/build/Lens.js:2:8665108)
```

**Comment by mateus4k** [comment]
Hi, @RotemBirman. Did you get another way to upgrade to v2beta2?

**Comment by RotemBirman** [comment]
Hey @mateus4k I was able to upgrade using rancher UI.
I beleieve that if I will do `kubectl apply -f hpa.yaml` that it will also work but did not have a chance to try it.

**Comment by mateus4k** [comment]
Thanks, @RotemBirman! Perhaps the error is external to Lens. I also tested it on Lens 5 and even though it didn't give an error, after saving the resource it goes back to v2beta1 again :thinking:

**Comment by RotemBirman** [comment]
@mateus4k to be fair my main issue was the fact that it goes back to v2beta1 and it won't show me the correct format of metrics + behavior section that is intreduced in v2beta2

**Comment by mateus4k** [comment]
Exactly!

**Comment by Nokel81** [comment]
Okay so this is actually a really difficult problem to solve. Because there is no definitive version within kube for the built-in types. 

Some background reading: https://github.com/kubernetes/kubernetes/issues/58131

You can verify this yourself by doing the following (assuming your hpa is named `foo` in namespace `bar`):

```
kubectl get hpa.v2beta2.autoscaling foo --namespace bar -o yaml
kubectl get hpa.v2beta1.autoscaling foo --namespace bar -o yaml
kubectl get hpa.v1.autoscaling foo --namespace bar -o yaml
```

you will get the same object back (`.metadata.uid` is the same).

**Comment by Nokel81** [comment]
This has been fixed by #6971
